axon-server-connector/src/main/java/org/axonframework/axonserver/connector/ServerConnectorConfigurationEnhancer.java

```diff
@@ -57,7 +59,23 @@ private <C> void registerIfNotPresent(ComponentRegistry registry,
         }
     }
 
-    private AxonServerConnectionManager buildAxonServerConnectionManager(Configuration config) {
+    // TODO this will be replaced by logic on the ComponentRegistry in a follow-up PR
+    private <C> void registerIfNotPresentDef(ComponentRegistry registry,
+                                             Class<C> type,
+                                             ComponentDefinition<AxonServerConnectionManager> definition) {
+        if (!registry.hasComponent(type)) {
+            registry.registerComponent(definition);
+        }
+    }
+
+    private ComponentDefinition<AxonServerConnectionManager> connectionManagerDefinition() {
+        return ComponentDefinition.ofType(AxonServerConnectionManager.class)
+                                  .withBuilder(this::buildConnectionManager)
+                                  .onStart(Phase.INSTRUCTION_COMPONENTS, AxonServerConnectionManager::start)
+                                  .onStart(Phase.EXTERNAL_CONNECTIONS, AxonServerConnectionManager::shutdown);
```

Comment:
This `onStart` should probably be `onShutdown`. And that's why the tests don't succeed

Reply:
whoops, good catch :see_no_evil: